Documentation/leak-detection.md

```diff
@@ -0,0 +1,24 @@
+# Leak detection (poisoning)
+
+## Overview
+
+Source-build includes a mechanism for *poisoning* its input files, and then for checking for that poison after the build is complete.  This allows us to ensure that all output files were built during the build rather than copied directly from the input, which would be an illegal prebuilt usage.
+
+## Before the build
+
+Before the build, the MarkAndCatalogFiles task runs.  This does a few things:
+
+- Record the hash of every file in the source-build binary input directories (prebuilt and previously-source-built.  future expansion: reference-packages).  If the file is a zip, tarball, or nupkg, unpack it and do the same thing recursively.
+- For managed DLLs, either bare or in an archive, add a custom attribute that marks the file as poisoned.
+- For nupkgs, drop a `.poisoned` file.
+- Repack the poisoned assemblies and extra files, removing signatures so they don't fail verification.
```

Comment:
Is "removing signatures" specific to the NuGet packages or does this apply to assemblies as well?

Reply:
Specified that this is the nupkgs only.